test/unit/meta/traits/common_logical.cpp

```diff
@@ -0,0 +1,79 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+**/
+//==================================================================================================
+
+#include "test.hpp"
+#include "unit/module/core/logical_test.hpp"
+
+#include <eve/module/core.hpp>
+
+TTS_CASE("eve::common_logical on boolean x boolean")
+{
+  TTS_TYPE_IS((eve::common_logical_t<bool, bool>), bool);
+};
+
+TTS_CASE_TPL("eve::common_logical on boolean x other", eve::test::simd::all_types)
+<typename T>(tts::type<T>)
+{
+  using S = eve::element_type_t<T>;
+
+  TTS_TYPE_IS((eve::common_logical_t<bool, T>), eve::logical<T>);
+  TTS_TYPE_IS((eve::common_logical_t<T, bool>), eve::logical<T>);
+  TTS_TYPE_IS((eve::common_logical_t<bool, eve::logical<T>>), eve::logical<T>);
+  TTS_TYPE_IS((eve::common_logical_t<eve::logical<T>, bool>), eve::logical<T>);
+
+  TTS_TYPE_IS((eve::common_logical_t<bool, S>), eve::logical<S>);
+  TTS_TYPE_IS((eve::common_logical_t<S, bool>), eve::logical<S>);
+  TTS_TYPE_IS((eve::common_logical_t<bool, eve::logical<S>>), eve::logical<S>);
+  TTS_TYPE_IS((eve::common_logical_t<eve::logical<S>, bool>), eve::logical<S>);
+};
+
+TTS_CASE_TPL("eve::common_logical on scalar x scalar", tts::cartesian_square<eve::test::scalar::all_types>)
+<typename T, typename U>( tts::type<kumi::tuple<T,U>> )
+{
+  using expected_t = priority_t<T, U>;
```

Comment:
Meh. You kinda do the same thing twice. I'd probably be explicit about different types. You don't have to list all of them.